[a11y] [com_fields] icons in modals

brianteeman opened the pull request:
Continuing the work to ensure that screen readers dont try to read the value of a font icon.

This PR could be improved with the use of titles as well if anyone has some suggestions to get the requires strings

brianteeman committed:
[a11y] icons in modals

Continuing the work to ensure that screen readers dont try to read the value of a font icon.

This PR could be improved with the use of titles as well if anyone has some suggestions to get the requires strings

Quy commented on administrator/components/com_categories/views/categories/tmpl/modal.php:
```diff
@@ -107,7 +107,7 @@
 						<tr class="row<?php echo $i % 2; ?>">
 							<td class="center">
-								<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Revert tab deletion.

brianteeman replied:
good spot thanks - fixed

brianteeman committed:
put tab back

tonypartridge commented:
I have tested this item :white_check_mark: successfully on 96e15759d9f8c9902b53f3999f2812e65b923bfd<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15047">issues.joomla.org/tracker/joomla-cms/15047</a>.</sub>

Quy commented:
Please provide instructions on how to invoke these modals. Thanks.

Quy commented:
```
PHP Notice:  Undefined property: stdClass::$published in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122

PHP Notice:  Undefined index:  in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122
```

Quy commented on administrator/components/com_content/views/articles/tmpl/modal.php:
```diff
@@ -119,7 +119,7 @@
 					<tr class="row<?php echo $i % 2; ?>">
 						<td class="center">
-							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
```
PHP Notice:  Undefined property: stdClass::$published in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122

PHP Notice:  Undefined index:  in \administrator\components\com_content\views\articles\tmpl\modal.php on line 122
```

Quy commented:
Here is an idea to get the status strings:
```
				<?php
				$iconStates = array(
					-2 => array('icon-trash', JText::_('JTRASHED')),
					0 => array('icon-unpublish', JText::_('JUNPUBLISHED')),
					1 => array('icon-publish', JText::_('JPUBLISHED')),
					2 => array('icon-archive', JText::_('JARCHIVED')),
				);
				?>
```
```
						<td class="center">
							<span class="<?php echo $iconStates[$this->escape($item->state)][0]; ?>" aria-hidden="true"></span> <span class="element-invisible"><?php echo $iconStates[$this->escape($item->state)][1]; ?></span>
						</td>
```

Quy commented on administrator/components/com_associations/views/associations/tmpl/modal.php:
```diff
@@ -106,7 +106,7 @@
 					<?php if (!empty($this->typeSupports['state'])) : ?>
 						<td class="center">
-							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Revert.

brianteeman replied:
oops - copy paste error

Quy commented on administrator/components/com_fields/views/fields/tmpl/modal.php:
```diff
@@ -91,7 +91,7 @@
 					<tr class="row<?php echo $i % 2; ?>">
 						<td class="center">
-							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
+							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
```
Revert `$iconStates[$this->escape($item->published)]`.

brianteeman committed:
Update modal.php

brianteeman committed:
changes

brianteeman commented:
typos fixed - thanks
@quy I will leave that suggestion for another PR

Quy commented:
I have tested this item :white_check_mark: successfully on 5b2b30a9e180d7db78834e6964f75d7b8522d119<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15047">issues.joomla.org/tracker/joomla-cms/15047</a>.</sub>

franz-wohlkoenig commented:
@tonypartridge can you please retest?

Commit:
Update article.xml (#16256)

* Update article.xml

* addtional XML CS fixes

* part 2

brianteeman committed:
[a11y] misc admin modules (#15068)

* logout button on mod_logged

* multilingual status icon

* admin stats modules line icons

* admin status module

* home icon on default menu

* update

I went for the `<span class="element-invisible">` instead of the aria-title so it is more consistent with other uses elsewhere

brianteeman commented:
Updated branch - no need for new tests

brianteeman committed:
Merge branch 'staging' into a11y_modal

rdeutz committed:
Merge branch 'staging' into a11y_modal

brianteeman committed:
Merge branch 'staging' into a11y_modal

tonypartridge commented:
anyone have any idea why AppVayor is stuck on waiting? @wilsonge ?

zero-24 committed:
Merge branch 'staging' into a11y_modal

zero-24 commented:
I have just gived it another try.

wilsonge commented:
Merged on review

brianteeman commented:
Thanks
